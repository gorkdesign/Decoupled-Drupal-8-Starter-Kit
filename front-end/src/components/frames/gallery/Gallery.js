import React from 'react';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';

import Thumbnail from './Thumbnail';
import BrowseButton from './BrowseButton';

const Gallery = ({
  onDrop,
  uploading,
  files,
  totalBytes,
  uploadInitiated = false,
  onUploadClick,
  handleCancel,
  handleDelete,
}) => {
  let dropzoneRef;

  return (
    <div className="GalleryFrameContainer">

      { uploading === false ?
        <Dropzone
          ref={(node) => { dropzoneRef = node; }}
          onDrop={onDrop}
          id="dropZone"
          className="dropZone"
          disabled={uploading}
        >
          <p>Drop your files here or click to browse.</p>
        </Dropzone>
      :
        <div id="dropZone" className="dropZone disabled">
          <p>Your files are uploading.</p>
        </div>
      }

      {uploading === false ?
        <BrowseButton
          files={files}
          totalBytes={totalBytes(files)}
          totalFiles={files.length}
          render={() => (
            <button
              type="button"
              onClick={() => { dropzoneRef.open(); }}
              disabled={uploadInitiated}
            >
            Choose Files
            </button>
        )}
        />
      : null }

      { files.length > 0 ?
        <button className="btn btn-primary" onClick={onUploadClick}>UPLOAD FILES</button>
      : null }

      <output id="list" className="container p-0">
        <div className="grid">
          {
            files.map((item, i) => {
              const image = files[i].file;
              if (image) {
                return (<Thumbnail
                  key={i}
                  handleCancel={handleCancel}
                  handleDelete={handleDelete}
                  index={i}
                  fileSize={image.size || image.fileSize}
                  fileName={image.name}
                  percentageComplete={image.percentCompleted ? image.percentCompleted : 0}
                  uploadInitiated={image.uploadInitiated ? image.uploadInitiated : false}
                  uploadSuccess={image.uploadSuccess ? image.uploadSuccess : false}
                  render={() => (
                    <figure>
                      <img alt="" src={item.thumbnail} className="responsive-image" />
                    </figure>
                  )}
                />);
              }
              return null;
            })
          }
        </div>
      </output>
    </div>
  );
};

Gallery.propTypes = {
  uploading: PropTypes.bool.isRequired,
  files: PropTypes.array.isRequired,
  uploadInitiated: PropTypes.bool,
  onDrop: PropTypes.func.isRequired,
  totalBytes: PropTypes.func.isRequired,
  onUploadClick: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

Gallery.defaultProps = {
  uploadInitiated: false,
};

export default Gallery;
