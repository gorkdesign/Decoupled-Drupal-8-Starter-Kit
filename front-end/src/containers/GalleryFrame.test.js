import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import createSagaMiddleware from 'redux-saga';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Gallery from './GalleryFrame';


describe('GalleryFrame', () => {
  const sagaMiddleware = createSagaMiddleware();
  const mockStore = configureMockStore([sagaMiddleware]);
  const store = mockStore({});


  const pageModel = {
    author: {
      name: 'admin',
    },
    title: 'hello article update',
    body: null,
    nid: 13,
    uuid: '79502776-61f8-4c48-b464-d94eebe0e01b',
    images: [],
  };

  const initialPropsState = {
    page: pageModel,
  };

  let props;
  let mountedLogin;
  const GalleryFrame = (modifiedProps = initialPropsState) => {
    props = {
      ...initialPropsState,
      ...modifiedProps,
    };

    if (!mountedLogin) {
      mountedLogin = mount(<Gallery {...props} />);
    }
    return mountedLogin;
  };

  beforeEach(() => {
    props = initialPropsState;
    mountedLogin = undefined;
  });

  describe('Snapshots', () => {
    const output = shallow(<Provider store={store}><GalleryFrame {...initialPropsState} /></Provider>);
    it('should render correctly', () => {
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });


  // describe('GalleryFrameArticle differing props', () => {
  //   it('Should NOT display the login form if authenticated - SHOULD REDIRECT', () => {
  //     const newProps = { ...initialPropsState };
  //     newProps.isAuthenticated = true;
  //     const wrapper = shallow(<Login {...newProps} />);
  //     expect(wrapper.find(Redirect).props().to).toBe('/');
  //   });
  // });
});
