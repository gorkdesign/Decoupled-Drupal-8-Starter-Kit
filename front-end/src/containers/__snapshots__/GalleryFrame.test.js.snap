// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryFrame Snapshots should render correctly 1`] = `
<GalleryFrame
  page={
    Object {
      "author": Object {
        "name": "admin",
      },
      "body": null,
      "images": Array [],
      "nid": 13,
      "title": "hello article update",
      "uuid": "79502776-61f8-4c48-b464-d94eebe0e01b",
    }
  }
/>
`;
